src/jit/codegencommon.cpp

```diff
@@ -3485,7 +3479,11 @@ void CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg, bool* pXtraRegClobbere
         {
             // A struct might be passed  partially in XMM register for System V calls.
             // So a single arg might use both register files.
-            if (isFloatRegType(regType) != doingFloat)
+            if (isFloatRegType(regType) != doingFloat
+#if defined(_TARGET_WINDOWS_) && defined(_TARGET_ARM64_)
+                && !compiler->info.compIsVarArgs
+#endif // defined(_TARGET_WINDOWS_) && defined(_TARGET_ARM64_)
+                )
```

Comment:
I'm confused by this. Unless I misunderstand, at this point `regType` should be the actual reg type that we're going to use (i.e. it's been mangled above as necessary for varargs), so I don't understand why the additional check is needed.
Further, I don't understand why the condition in `Compiler::mangleVarArgsType` isn't conditional on `_TARGET_WINDOWS_`. I definitely have the feeling I'm confused here, but I think the upshot is that this and the `Compiler::mangleVarArgsType` need additional comments.
Also, I would break this up for clarity, e.g.:
```
// Is this arg going into the register file we're currently processing?
bool handleThisArg = (isFloatRegType(regType) == doingFloat);

#if defined(_TARGET_WINDOWS_) && defined(_TARGET_ARM64_)
// <Suitable comment here about why this additional condition is required for ARM64/windows.>
handleThisArg ||= compiler->info.compIsVarArgs;
+#endif // defined(_TARGET_WINDOWS_) && defined(_TARGET_ARM64_)

if (!handleThisArg)
{
    continue;
}
```

Reply:
mangleVarArgs is unfortunately incorrectly implemented. It was modified by the softFP work to retype on armel. The current name should be mangleVarArgsAndSoftFPTypes, I would like to change this eventually to split the function into two seperate methods. As native varargs on unix arm(64) will take this codepath incorrectly.

Reply:
Also note that the method relies on compIsVarArgs, so it can only be used for incoming callee arguments. fgMorphArgs cannot rely on the method, which also seems to be a fundamental flaw.

Reply:
> mangleVarArgs is unfortunately incorrectly implemented. It was modified by the softFP work to retype on armel. The current name should be mangleVarArgsAndSoftFPTypes, I would like to change this eventually to split the function into two seperate methods.

Maybe I haven't read far enough, but I don't understand this comment. It was done as is because SoftFP should behave exactly as varargs, where everything gets passed in integer registers on arm(el). Is that not true anymore?

Reply:
You are correct, the change is a layover from when I was testing without mangling the type. Thank you for catching this.

Reply:
I also don't see why we need the additional condition here (` && !compiler->info.compIsVarArgs`). It seems like having it would cause the floating-point arguments in a varargs function to get processed twice: once with the integer register file, and once with the floating-point register file.